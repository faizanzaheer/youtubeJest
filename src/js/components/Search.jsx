import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

export default class Search extends React.Component {
  constructor() {
    super();
    this.state = {
      value: ""
    }

    this.localHandler = this.localHandler.bind(this)
    // this.localHandler = _.debounce(this.localHandler, 500)
  }

  localHandler(e) {
    const { getVideos } = this.props;
    // run the api with the input value
    this.setState({value: e.target.value})
    getVideos(e.target.value);

  }

  render() {
    return (
      <div className="search col-md-6 col-md-offset-4 form-inline"> 
        <input 
          type="text" 
          className="form-control"
          // ref={(elem)=>this.input=elem}
          onChange={this.localHandler}
          value = {this.state.value}
        />
        <button className="btn">
          <span className="glyphicon glyphicon-search "></span>
        </button>
      </div>
    )
  }
};

Search.propTypes = {
  getVideos: PropTypes.func.isRequired,
}








