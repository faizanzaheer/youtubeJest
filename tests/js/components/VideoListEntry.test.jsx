import React from 'react';
import  {VideoListEntry}  from '../../../src/js/components/VideoListEntry';


import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme'
Enzyme.configure({ adapter: new Adapter() });

import renderer from 'react-test-renderer';
//this package is provided by jest(replacing react-test)
describe('Testing VideoListEntry....', ()=>{
	const video = {
		snippet: {
			thumbnails:{
				default:{
					url: 'youtube.com'
				}
			},title: 'video snippet title',
			description: 'video snippet description'

			}
		}
		const handleSelectedVideo = jest.fn();
		const wrapper = shallow(<VideoListEntry
			video = {video}
			handleSelectedVideo={handleSelectedVideo}/>);
		it('renders the VideoListEntry companent',()=>{
			expect(wrapper).toMatchSnapshot();
		});
		it('testing selective function', ()=>{
			expect(handleSelectedVideo.mock.calls.length).toBe(0);
		});
		it('handleSelectedVideo call..',()=>{
			wrapper.find('.media-heading').simulate('click')
			expect(handleSelectedVideo.mock.calls[0][0]).toEqual(video)
		});
	});
