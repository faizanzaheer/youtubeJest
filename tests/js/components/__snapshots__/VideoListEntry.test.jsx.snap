// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoListEntry.... renders the VideoListEntry companent 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    video={
      Object {
        "snippet": Object {
          "description": "video snippet description",
          "thumbnails": Object {
            "default": Object {
              "url": "youtube.com",
            },
          },
          "title": "video snippet title",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="media"
      >
        <div
          className="media-left media-middle"
        >
          <img
            className="media-object"
            src="youtube.com"
          />
        </div>
        <div
          className="media-body"
        >
          <h4
            className="media-heading"
            onClick={[Function]}
          >
            video snippet title
          </h4>
          <p
            className="media-description"
          >
            video snippet description
          </p>
        </div>
      </div>,
      "className": "videoListEntry",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="media-left media-middle"
          >
            <img
              className="media-object"
              src="youtube.com"
            />
          </div>,
          <div
            className="media-body"
          >
            <h4
              className="media-heading"
              onClick={[Function]}
            >
              video snippet title
            </h4>
            <p
              className="media-description"
            >
              video snippet description
            </p>
          </div>,
        ],
        "className": "media",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="media-object"
              src="youtube.com"
            />,
            "className": "media-left media-middle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "media-object",
              "src": "youtube.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="media-heading"
                onClick={[Function]}
              >
                video snippet title
              </h4>,
              <p
                className="media-description"
              >
                video snippet description
              </p>,
            ],
            "className": "media-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "video snippet title",
                "className": "media-heading",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "video snippet title",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "video snippet description",
                "className": "media-description",
              },
              "ref": null,
              "rendered": "video snippet description",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="media"
        >
          <div
            className="media-left media-middle"
          >
            <img
              className="media-object"
              src="youtube.com"
            />
          </div>
          <div
            className="media-body"
          >
            <h4
              className="media-heading"
              onClick={[Function]}
            >
              video snippet title
            </h4>
            <p
              className="media-description"
            >
              video snippet description
            </p>
          </div>
        </div>,
        "className": "videoListEntry",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="media-left media-middle"
            >
              <img
                className="media-object"
                src="youtube.com"
              />
            </div>,
            <div
              className="media-body"
            >
              <h4
                className="media-heading"
                onClick={[Function]}
              >
                video snippet title
              </h4>
              <p
                className="media-description"
              >
                video snippet description
              </p>
            </div>,
          ],
          "className": "media",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="media-object"
                src="youtube.com"
              />,
              "className": "media-left media-middle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "media-object",
                "src": "youtube.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="media-heading"
                  onClick={[Function]}
                >
                  video snippet title
                </h4>,
                <p
                  className="media-description"
                >
                  video snippet description
                </p>,
              ],
              "className": "media-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "video snippet title",
                  "className": "media-heading",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "video snippet title",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "video snippet description",
                  "className": "media-description",
                },
                "ref": null,
                "rendered": "video snippet description",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
